src/renderer/families/polkadot/ManageModal.js

```diff
@@ -0,0 +1,253 @@
+  background-color: rgba(0, 0, 0, 0);
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: flex-start;
+
+  &:hover {
+    border: 1px solid ${p => p.theme.colors.palette.primary.main};
+  }
+
+  ${p =>
+    p.disabled
+      ? css`
+          pointer-events: none;
+          cursor: auto;
+          ${IconWrapper} {
+            background-color: ${p.theme.colors.palette.action.active};
+            color: ${p.theme.colors.palette.text.shade20};
+          }
+          ${Title} {
+            color: ${p.theme.colors.palette.text.shade50};
+          }
+          ${Description} {
+            color: ${p.theme.colors.palette.text.shade30};
+          }
+        `
+      : `
+      cursor: pointer;
+  `};
+`;
+
+const InfoWrapper = styled(Box).attrs(() => ({
+  vertical: true,
+  flex: 1,
+  ml: 3,
+  textAlign: "start",
+}))``;
+
+const Title = styled(Text).attrs(() => ({
+  ff: "Inter|SemiBold",
+  fontSize: 4,
+}))``;
+
+const Description = styled(Text).attrs(({ isPill }) => ({
+  ff: isPill ? "Inter|SemiBold" : "Inter|Regular",
+  fontSize: isPill ? 2 : 3,
+  color: "palette.text.shade60",
+}))`
+  ${p =>
+    p.isPill
+      ? `
+    text-transform: uppercase;
+  `
+      : ""}
+`;
+
+type Props = {
+  name?: string,
+  account: Account,
+  ...
+};
+
+const ManageModal = ({ name, account, ...rest }: Props) => {
+  const dispatch = useDispatch();
+
+  const { staking } = usePolkadotPreloadData();
+
+  const { polkadotResources } = account;
+
+  invariant(polkadotResources, "polkadot account expected");
+
+  const { unlockedBalance, nominations } = polkadotResources;
+
+  const onSelectAction = useCallback(
+    (onClose, name, params = {}) => {
+      onClose();
+      dispatch(
+        openModal(name, {
+          account,
+          ...params,
+        }),
+      );
+    },
+    [dispatch, account],
+  );
+
+  const electionOpen = staking?.electionClosed !== undefined ? !staking?.electionClosed : false;
+  const hasUnlockedBalance = unlockedBalance && unlockedBalance.gt(0);
+
+  const nominationEnabled = !electionOpen && canNominate(account);
+  const chillEnabled = !electionOpen && canNominate(account) && nominations?.length;
+  const bondingEnabled = !electionOpen && canBond(account);
+  const unbondingEnabled = !electionOpen && canUnbond(account);
+  const withdrawEnabled = !electionOpen && hasUnlockedBalance;
+
+  return (
+    <Modal
+      {...rest}
+      name={name}
+      centered
+      render={({ onClose, data }) => (
+        <ModalBody
+          onClose={onClose}
+          onBack={undefined}
+          title={<Trans i18nKey="polkadot.manage.title" />}
+          noScroll
+          render={() => (
+            <>
+              <Box>
+                {electionOpen ? <ElectionStatusWarning /> : null}
+                <ManageButton
+                  disabled={!bondingEnabled}
+                  onClick={() => onSelectAction(onClose, "MODAL_POLKADOT_BOND")}
```

Comment:
curried function don't help on perf (i mean usecallback won't cache the function you recreate on calling it.

howeevr, 

```
() => onSelectAction(onClose, "MODAL_POLKADOT_BOND")
```

can be in a useCallback

Reply:
So it would be a useCallback inception ?

Reply:
you cannot dynamically call a useCallback as the hooks calls count must be fixed and never changing (same count and same order).

Note that this is how it's implemented in other coins.

Reply:
But...

We cannot useCallback in the click handlers, and we have to pass onClose as new arrow function anyway, since it's provided by the <ModalBody>.

The only way to avoid this is to pass a ref to ModalBody using a useRef for the onClose handler, but this would require a change of the ModalBody component, and IMHO it's a really weird, dangerous and bad practice / anti-pattern.